### DISCUSSION TITLE
Uploading files to sd card not working

### DISCUSSION THREAD

**Discussion by SebastianMusser** [grblHAL/core#520] [General]
Hi Terje,
I run a flexihal board with the sd/eth extension and the latest firmware. 
Using IOsender 2.0.44, I can **read** the files from the sd card and I can also **delete** files, BUT I **cannot upload** anything.
When I try to do that, IOsender always says "Unable to connect to remote server".

Using ftp via eth works fine.

Any ideas if I am doing something wrong?

Thx!

**Comment by terjeio** [comment]
Are you running the ftp server on a different port that 21?

**Reply by SebastianMusser** [reply]
Morning Terjeio!
Sorry - my initial posting wasn't accurate enough. I am trying to copy files in IOsender using USB only, NOT with ethernet / ftp. 

However, I did not change the ftp port and i can connect via ftp (ftp://192.168.5.1:21) when I use an additional ethernet cable. 


Regards!

**Reply by terjeio** [reply]
Well, ioSender uses YModem/serial to upload files when SDCARD_ENABLE is set to 2. But if FTP is available that is used instead since it is a lot faster. The error message you get is from trying to connect to the FTP server.
Perhaps I should fall back to YModem if FTP fails...

**Reply by SebastianMusser** [reply]
Awesome, disabling ftp did the trick. 

Didn’t think of that myself as downloading and deleting files worked. 

Thx!
